File: Sample/Sample/Lua/Items/Button.lua

hlayout = Gwen.HorizontalLayout.Create(Gwen.Canvas)

	vlayout = Gwen.VerticalLayout.Create(hlayout)
	vlayout.Width = 300

		button = Gwen.Button.Create(vlayout)
		button.Margin = Gwen.Margin.Create(5)
		button.Text = "Button"

		button = Gwen.Button.Create(vlayout)
		button.Margin = Gwen.Margin.Create(5)
		button.Padding = Gwen.Padding.Create(3)
		button.Text = "Image button (default)"
		button.ImageName = "test16.png"

		button = Gwen.Button.Create(vlayout)
		button.Margin = Gwen.Margin.Create(5)
		button.Padding = Gwen.Padding.Create(3)
		button.Text = "Image button (above)"
		button.ImageName = "test16.png"
		button.ImageAlign = Gwen.ImageAlign.Above

		button = Gwen.Button.Create(vlayout)
		button.Margin = Gwen.Margin.Create(5)
		button.Padding = Gwen.Padding.Create(3)
		button.Alignment = Gwen.Alignment.Left .. Gwen.Alignment.CenterV
		button.Text = "Image button (left)"
		button.ImageName = "test16.png"

		button = Gwen.Button.Create(vlayout)
		button.Margin = Gwen.Margin.Create(5)
		button.Padding = Gwen.Padding.Create(3)
		button.Alignment = Gwen.Alignment.Right .. Gwen.Alignment.CenterV
		button.Text = "Image button (right)"
		button.ImageName = "test16.png"

		button = Gwen.Button.Create(vlayout)
		button.Margin = Gwen.Margin.Create(5)
		button.Padding = Gwen.Padding.Create(3)
		button.Text = "Image button (image left)"
		button.ImageName = "test16.png"
		button.ImageAlign = Gwen.ImageAlign.Left .. Gwen.ImageAlign.CenterV

		button = Gwen.Button.Create(vlayout)
		button.Margin = Gwen.Margin.Create(5)
		button.Padding = Gwen.Padding.Create(3)
		button.Text = "Image button (image right)"
		button.ImageName = "test16.png"
		button.ImageAlign = Gwen.ImageAlign.Right .. Gwen.ImageAlign.CenterV

		button = Gwen.Button.Create(vlayout)
		button.Margin = Gwen.Margin.Create(5)
		button.Padding = Gwen.Padding.Create(3)
		button.Text = "Image button (image fill)"
		button.ImageName = "test16.png"
		button.ImageAlign = Gwen.ImageAlign.Fill

--[[
			HorizontalLayout hlayout = new HorizontalLayout(this);
			{
				VerticalLayout vlayout = new VerticalLayout(hlayout);
				vlayout.Width = 300;
				{


					button = new Control.Button(vlayout);
					button.Margin = Margin.Five;
					button.Padding = Padding.Three;
					button.Text = "Image button (image fill)";
					button.SetImage("test16.png", ImageAlign.Fill);

					HorizontalLayout hlayout2 = new HorizontalLayout(vlayout);
					{
						button = new Control.Button(hlayout2);
						button.HorizontalAlignment = HorizontalAlignment.Left;
						button.Padding = Padding.Three;
						button.Margin = Margin.Five;
						button.SetImage("test16.png");
						button.ImageSize = new Size(32, 32);

						button = new Control.Button(hlayout2);
						button.HorizontalAlignment = HorizontalAlignment.Left;
						button.VerticalAlignment = VerticalAlignment.Center;
						button.Padding = Padding.Three;
						button.Margin = Margin.Five;
						button.SetImage("test16.png");

						button = new Control.Button(hlayout2);
						button.HorizontalAlignment = HorizontalAlignment.Left;
						button.VerticalAlignment = VerticalAlignment.Center;
						button.Padding = Padding.Three;
						button.Margin = Margin.Five;
						button.SetImage("test16.png");
						button.ImageTextureRect = new Rectangle(4, 4, 8, 8);

						button = new Control.Button(hlayout2);
						button.HorizontalAlignment = HorizontalAlignment.Left;
						button.VerticalAlignment = VerticalAlignment.Center;
						button.Padding = Padding.Three;
						button.Margin = Margin.Five;
						button.SetImage("test16.png");
						button.ImageColor = Color.DarkGrey;
					}

					button = new Control.Button(vlayout);
					button.Margin = Margin.Five;
					button.Padding = new Padding(20, 20, 20, 20);
					button.Text = "Toggle me";
					button.IsToggle = true;
					button.Toggled += onToggle;
					button.ToggledOn += onToggleOn;
					button.ToggledOff += onToggleOff;

					button = new Control.Button(vlayout);
					button.Margin = Margin.Five;
					button.Padding = Padding.Three;
					button.Text = "Disabled";
					button.IsDisabled = true;

					button = new Control.Button(vlayout);
					button.Margin = Margin.Five;
					button.Padding = Padding.Three;
					button.Text = "With Tooltip";
					button.SetToolTipText("This is tooltip");

					button = new Control.Button(vlayout);
					button.Margin = Margin.Five;
					button.Padding = Padding.Three;
					button.Text = "Autosized";
					button.HorizontalAlignment = HorizontalAlignment.Left;
				}

				{
					Control.Button button = new Control.Button(hlayout);
					button.Margin = Margin.Five;
					button.Padding = Padding.Three;
					button.Text = "Event tester";
					button.Size = new Size(300, 200);
					button.Pressed += onButtonAp;
					button.Clicked += onButtonAc;
					button.Released += onButtonAr;
				}
			}
--]]












AddItem("Standard", "Button", hlayout)

File: Library/Portable/Control/Button.cs
using System;
using MoonSharp.Interpreter;

namespace GwenNetLua.Control
{
	public class Button : Internal.ButtonBase
	{
		public static Button Create(Internal.ControlBase parent)
		{
			return new Button(new Gwen.Control.Button(parent.Target));
		}

		[MoonSharpHidden]
		public static Button Create(Gwen.Control.ControlBase control)
		{
			if (control == null)
				return null;
			else
				return new Button(control);
		}

		[MoonSharpHidden]
		public Button(Gwen.Control.ControlBase control)
			: base(control)
		{
		}

		public string Text { get { return GetTarget<Gwen.Control.Button>().Text; } set { GetTarget<Gwen.Control.Button>().Text = value; } }
		public Gwen.Alignment Alignment { get { return GetTarget<Gwen.Control.Button>().Alignment; } set { GetTarget<Gwen.Control.Button>().Alignment = value; } }
		public Font Font { get { return new Font(GetTarget<Gwen.Control.Button>().Font); } set { GetTarget<Gwen.Control.Button>().Font = value.Target; } }
		public Color TextColor { get { return new Color(GetTarget<Gwen.Control.Button>().TextColor); } set { GetTarget<Gwen.Control.Button>().TextColor = value.Target; } }
		public Color TextColorOverride { get { return new Color(GetTarget<Gwen.Control.Button>().TextColorOverride); } set { GetTarget<Gwen.Control.Button>().TextColorOverride = value.Target; } }
		public Padding TextPadding { get { return new Padding(GetTarget<Gwen.Control.Button>().TextPadding); } set { GetTarget<Gwen.Control.Button>().TextPadding = value.Target; } }
		public string ImageName { get { return GetTarget<Gwen.Control.Button>().ImageName; } set { GetTarget<Gwen.Control.Button>().ImageName = value; } }
		public Gwen.Control.ImageAlign ImageAlign { get { return GetTarget<Gwen.Control.Button>().ImageAlign; } set { GetTarget<Gwen.Control.Button>().ImageAlign = value; } }
		public Color ImageColor { get { return new Color(GetTarget<Gwen.Control.Button>().ImageColor); } set { GetTarget<Gwen.Control.Button>().ImageColor = value.Target; } }
		public Size ImageSize { get { return new Size(GetTarget<Gwen.Control.Button>().ImageSize); } set { GetTarget<Gwen.Control.Button>().ImageSize = value.Target; } }
		public Rectangle ImageTextureRect { get { return new Rectangle(GetTarget<Gwen.Control.Button>().ImageTextureRect); } set { GetTarget<Gwen.Control.Button>().ImageTextureRect = value.Target; } }

		public void SetImage(string imageName, Gwen.Control.ImageAlign imageAlign = Gwen.Control.ImageAlign.LeftSide) { GetTarget<Gwen.Control.Button>().SetImage(imageName, imageAlign); }
	}
}
